src/components/contract-management/ContractManagementList.tsx

```diff
@@ -236,6 +240,233 @@ class ContractManagementList extends React.Component<Props, State> {
       />
     );
   }
+
+  /**
+   * Method for rendering edit buttons
+   */
+  private renderEditButtons = () => {
+    const { tableEditMode } = this.state;
+
+    if (!tableEditMode) {
+      return (
+        <Button onClick={this.toggleTableEditMode}>Muokkaustila</Button>
+      );
+    }
+
+    return (
+      <React.Fragment>
+        <Button onClick={this.saveTable}>Tallenna</Button>
+        <Button onClick={this.toggleTableEditMode}>Peruuta</Button>
+      </React.Fragment>
+    );
+  }
+
+  /**
+   * Method for rendering contract status
+   *
+   * @param contract contract
+   */
+  private renderEditableStatus = (contract: Contract) => {
+    const { tableEditMode } = this.state;
+
+    if (!tableEditMode) {
+      return contract.status;
+    }
+
+    const { editedContracts } = this.state;
+    const statusOptions = this.getStatusOptions();
+    const editedContract = editedContracts.find(item => item.id === contract.id) || contract;
+    const value = editedContract.status;
+    
+    return (
+      <Dropdown
+        fluid
+        selection
+        value={value}
+        options={statusOptions}
+        onChange={this.editContractStatus(editedContract)}
+      />
+    );
+  }
+
+  /**
+   * Method for rendering contract quantity
+   *
+   * @param contract contract
+   */
+  private renderEditableQuantity= (contract: Contract) => {
+    const { tableEditMode } = this.state;
+
+    if (!tableEditMode) {
+      return contract.contractQuantity;
+    }
+
+    const { editedContracts } = this.state;
+    const editedContract = editedContracts.find(item => item.id === contract.id) || contract;
+    const value = editedContract.contractQuantity;
+
+    return (
+      <Input
+        fluid
+        value={value}
+        onChange={this.editContractQuantity(editedContract)}
+      />
+    );
+  }
+
+  /**
+   * Method for rendering contract remark
+   *
+   * @param contract contract
+   */
+  private renderEditableRemark = (contract: Contract) => {
+    const { tableEditMode } = this.state;
+
+    if (!tableEditMode) {
+      return contract.remarks;
+    }
+
+    const { editedContracts } = this.state;
+    const editedContract = editedContracts.find(item => item.id === contract.id) || contract;
+    const value = editedContract.remarks;
+
+    return (
+      <TextArea
+        fluid
+        value={value}
+        onChange={this.editContractRemark(editedContract)}
+      />
+    );
+  }
+
+  /**
+   * Method for editing contract status
+   *
+   * @param contract contract
+   */
+  private editContractStatus = (contract: Contract) => (event: React.ChangeEvent<any>, data: DropdownProps) => {
+    const { editedContracts } = this.state;
+
+    const value = this.getStatus(String(data.value));
+
+    if (editedContracts.includes(contract) && value) {
+      contract.status = value;
+      this.setState({ editedContracts: [...editedContracts] });
```

Comment:
I figured it would be unnecessary since incase a contract is already in the editedContracts array then the contract can be mutated via reference. Mutating doesn't trigger render which is why editedContracts is spreaded in the set state function. Should I change this?